app/frontend/src/features/profile/edit/EditHostingPreference.tsx

```diff
@@ -250,31 +252,25 @@ export default function HostingPreferenceForm() {
             defaultValue={user.smokingAllowed}
             name="smokingAllowed"
             render={({ onChange }) => (
-              <Autocomplete
-                disableClearable={false}
-                defaultValue={user.smokingAllowed}
-                forcePopupIcon
-                freeSolo={false}
-                getOptionLabel={(option) => smokingLocationLabels[option]}
-                multiple={false}
-                options={[
+              <Select
+                defaultValue={user.smokingAllowed || SmokingLocation.SMOKING_LOCATION_UNKNOWN}
+                onChange={onChange}
+                label={ACCEPT_SMOKING}
+                name="smokingAllowed"
+                className={classes.field}
+              >
+                {[
+                  SmokingLocation.SMOKING_LOCATION_UNKNOWN,
                   SmokingLocation.SMOKING_LOCATION_YES,
                   SmokingLocation.SMOKING_LOCATION_WINDOW,
                   SmokingLocation.SMOKING_LOCATION_OUTSIDE,
                   SmokingLocation.SMOKING_LOCATION_NO,
-                ]}
-                onChange={(e, value) =>
-                  onChange(value ?? SmokingLocation.SMOKING_LOCATION_UNKNOWN)
-                }
-                renderInput={(params) => (
-                  <ProfileTextInput
-                    {...params}
-                    label={ACCEPT_SMOKING}
-                    name="smokingAllowed"
-                    className={classes.field}
-                  />
-                )}
-              />
+                ].map((o) => (
+                  <option value={o} key={o}>
+                    {smokingLocationLabels[o]}
+                  </option>
```

Comment:
Instead of doing this every time, what do you think of giving `<Select>` an `options: {label: string, value: T}[]` prop and doing the map in the component?

Reply:
will do :)